File: src/main/java/me/nasukhov/TukitaLearner/bot/command/Handler.java
package me.nasukhov.TukitaLearner.bot.command;

import me.nasukhov.TukitaLearner.bot.io.Input;
import me.nasukhov.TukitaLearner.bot.io.Output;

public interface Handler {
    boolean supports(Input input);
    void handle(Input input, Output output);
}

File: src/main/kotlin/me/nasukhov/TukitaLearner/bot/command/Handler.kt
package me.nasukhov.TukitaLearner.bot.command

import me.nasukhov.TukitaLearner.bot.io.Input
import me.nasukhov.TukitaLearner.bot.io.Output

interface Handler {
    fun supports(input: Input): Boolean
    fun handle(input: Input, output: Output)
}
